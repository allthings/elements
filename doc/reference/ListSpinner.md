<!-- 
This is an auto-generated markdown. 
You can change it in "/Users/daniel/Dev/allthings/elements/src/ListSpinner/ListSpinner.tsx" and run build:docs to update this file.
-->
# ListSpinner
It's a spinner for a list *
## Usage
| Name        | Type           | Description  |
| ----------- |:--------------:| ------------:|
|radius **(required)**||
