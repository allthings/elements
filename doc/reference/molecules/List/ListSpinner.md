# ListSpinner
It's a spinner for a list *
## Usage
| Name        | Type           | Description  |
| ----------- |:--------------:| ------------:|
*No properties to pass*
