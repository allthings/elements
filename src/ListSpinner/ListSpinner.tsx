import React, { FunctionComponent } from 'react'
import View from '../View'
import Spinner from '../Spinner'
import Circle, { ICircleProps } from '../Circle'

/** It's a spinner for a list **/
const ListSpinner: FunctionComponent<IListSpinnerProps> = ({
  radius = 40,
  ...props
}) => (
  <View direction="row" alignH="center">
    <Circle radius={radius} color="white" {...props}>
      <Spinner size={radius - 10} />
    </Circle>
  </View>
)

interface IListSpinnerProps extends ICircleProps {
  readonly radius?: number
}

export default ListSpinner
