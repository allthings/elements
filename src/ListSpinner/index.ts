export { default } from './ListSpinner'
export * from './ListSpinner'
