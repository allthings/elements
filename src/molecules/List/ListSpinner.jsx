import React from 'react'
import View from '@allthings/react-view'
import Spinner from '../../atoms/Spinner'
import Circle from '../../atoms/Circle'

const ListSpinner = props => (
  <View direction="row" alignH="center">
    <Circle radius={40} color="white" {...props}>
      <Spinner />
    </Circle>
  </View>
)

export default ListSpinner
