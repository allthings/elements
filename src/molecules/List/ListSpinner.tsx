import React, { FunctionComponent } from 'react'
import View from '../../atoms/View'
import Spinner from '../../atoms/Spinner'
import Circle from '../../atoms/Circle'

/** It's a spinner for a list **/
const ListSpinner: FunctionComponent<IListSpinnerProps> = ({
  radius = 40,
  ...props
}) => (
  <View direction="row" alignH="center">
    <Circle radius={radius} color="white" {...props}>
      <Spinner size={radius - 10} />
    </Circle>
  </View>
)

interface IListSpinnerProps {
  radius: number
}

export default ListSpinner
