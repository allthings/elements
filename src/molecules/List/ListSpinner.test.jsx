import React from 'react'
import renderer from 'react-test-renderer'
import ListSpinner from './ListSpinner'

test('color passed down to ListSpinner correctly', () => {
  const tree = renderer
    .create(<ListSpinner color="#000000" radius={50} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
