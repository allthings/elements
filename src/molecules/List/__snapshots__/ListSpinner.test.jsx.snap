// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`color passed down to ListSpinner correctly 1`] = `
.css-0,
[data-css-0] {
  box-sizing: border-box;
  align-content: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  justify-content: center;
  flex: 0 0 auto;
  -webkit-align-content: stretch;
  -webkit-box-align: stretch;
  -webkit-align-items: stretch;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-flex: 0 0 auto;
}

.css-1,
[data-css-1] {
  background-color: white;
  border-radius: 50%;
  height: 50px;
  width: 50px;
  transition: background-color 120ms linear;
  -webkit-transition: background-color 120ms linear;
  -moz-transition: background-color 120ms linear;
}

.css-2,
[data-css-2] {
  box-sizing: border-box;
  align-content: center;
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  justify-content: center;
  flex: 0 0 auto;
  -webkit-align-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-flex: 0 0 auto;
}

<div
  data-css-0=""
>
  <div
    data-css-1=""
    data-css-2=""
  >
    <svg
      height={40}
      preserveAspectRatio="xMidYMid"
      viewBox="0 0 100 100"
      width={40}
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="none"
        height="100"
        width="100"
        x="0"
        y="0"
      />
      <circle
        cx="50"
        cy="50"
        fill="none"
        r="40"
        stroke="#ffffff"
        strokeLinecap="round"
        strokeWidth="10"
      />
      <circle
        cx="50"
        cy="50"
        fill="none"
        r="40"
        stroke="#000000"
        strokeLinecap="round"
        strokeWidth="8"
      >
        <animate
          attributeName="stroke-dashoffset"
          dur="2s"
          from="0"
          repeatCount="indefinite"
          to="502"
        />
        <animate
          attributeName="stroke-dasharray"
          dur="2s"
          repeatCount="indefinite"
          values="125.5 125.5;1 250;125.5 125.5"
        />
      </circle>
    </svg>
  </div>
</div>
`;
